bluepyemodel/emodel_pipeline/plotting.py

```diff
@@ -1372,49 +1415,103 @@ def plot_FI_curves_comparison(
     # pylint: disable=too-many-nested-blocks, possibly-used-before-assignment
     make_dir(figures_dir)
 
-    emodel_name = None
-    cells = None
-    for emodel in emodels:
-        # do not re-extract data if the emodel is the same as previously
-        if custom_bluepyefe_cells_pklpath is not None:
+    emodel_name, cells = None, None
+
+    def load_cells_for_emodel(emodel):
```

Comment:
Is there a reason you are defining a bunch of functions inside this function? Why not defining them in plotting_utils.py?

Reply:
The functions were originally defined inside because I was using a global variable with the earlier logic. Since I have changed the logic, they can now be defined outside. And moving them to plotting_utils.py is a good idea